lib/msf/core/db_manager/payload.rb

```diff
@@ -0,0 +1,53 @@
+module Msf::DBManager::Payload
+
+  def create_payload(opts)
+    if opts[:uuid] && !opts[:uuid].to_s.empty?
+      if Mdm::Payload.find_by(uuid: opts[:uuid])
+        raise ArgumentError.new("A payload with this uuid already exists.")
+      end
+    end
+
+    Mdm::Payload.create(opts)
+  end
+
+  def payloads(opts)
+    ::ActiveRecord::Base.connection_pool.with_connection do
+      if opts[:id] && !opts[:id].to_s.empty?
+        return Array.wrap(Mdm::Payload.find(opts[:id]))
+      end
+
+      opts.delete(:workspace)
```

Comment:
Other models use `Msf::Util::DBManager.process_opts_workspace` to ensure that `:workspace` is present. I'm not really sure that it's necessary here, but it does make it inconsistent with other models.